homeassistant/helpers/translation.py

```diff
@@ -313,17 +317,14 @@ def _build_category_cache(
             for component, resource in new_resources.items():
                 component_cache = category_cache.setdefault(component, {})
```

Comment:
wouldn't be better that instead of having the class' `self.cache` a normal dict, to have it as a `defaultdict` which avoids the need at the different levels to have to set another dictionary as the default value: https://docs.python.org/3/library/collections.html#collections.defaultdict

Reply:
Yes that would be a bit cleaner, feel free to open a PR to adjust after this one merges.

Reply:
The icon helper could likely use the same

Reply:
@albertomontesg Is this something you want to take on? If not I can add it to my queue.

Reply:
I'll have to say that this is quite minor quality change that hardly would make a difference on the existing codebase. Feel free to take it if you want.

Reply:
https://github.com/home-assistant/core/pull/115816